### PR TITLE
Add Store Update and Delete APIs

### PR DESCRIPTION
**Pull request by ZinnaChoi**
Hello @JoshRyu~

I've implemented store delete and update APIs!
I've also simplified the complex service logic.

Now, I've successfully addressed this [issue](https://github.com/JoshRyu/logistics/issues/7).
Please take a look and feel free to provide feedback!

Thank you.

### CODE REVIEW

File: backend/src/main/java/com/madeg/logistics/service/CategoryService.java [hunk 1]
```diff
@@ -70,49 +71,46 @@ public void patchCategory(String code, CategoryPatch patchInput) {
...
+    if (
+      !Objects.equals(
+        parentCategoryCode,
+        currentParentCategory != null
+          ? currentParentCategory.getCategoryCode()
+          : null
+      )
+    ) {
+      Category parentCategory = categoryRepository.findByCategoryCode(
+        patchInput.getParentCategoryCode()
```
Comment (JoshRyu) [inline backend/src/main/java/com/madeg/logistics/service/CategoryService.java:101]: you've declared parentCategoryCode in line 89, what about reusing it?

File: backend/src/main/java/com/madeg/logistics/service/StoreService.java [hunk 2]
```diff
@@ -41,4 +44,85 @@ public void createStore(StoreInput storeInput) {
...
+        HttpStatus.NOT_FOUND,
+        "STORE NOT FOUND"
+      );
+    }
+
+    boolean isUpdated = false;
+
+    if (!Objects.equals(patchInput.getName(), previousStore.getName())) {
+      previousStore.updateName(patchInput.getName());
+      isUpdated = true;
```
Comment (JoshRyu) [inline backend/src/main/java/com/madeg/logistics/service/StoreService.java:62]: seems isUpdated = true; statement is repeated throughout the code. 
can't we use different approach? for example, comparing object itself to validate whether its updated or not?

**Review by JoshRyu — CHANGES_REQUESTED** [review]
I have added few comments! please have a look at them!

File: backend/src/main/java/com/madeg/logistics/service/CategoryService.java (hunk 1 above)
Comment (ZinnaChoi) [reply backend/src/main/java/com/madeg/logistics/service/CategoryService.java]: I reuse the variable! thanks~

File: backend/src/main/java/com/madeg/logistics/service/StoreService.java (hunk 2 above)
Comment (ZinnaChoi) [reply backend/src/main/java/com/madeg/logistics/service/StoreService.java]: I use object change detection logic without using isUpdated flag
I changed for product, category and store!

check it and give me a feedback plz 😛

**Review by JoshRyu — APPROVED** [review]
Looks good!

**Pull request merged**